content/languages/python-poetry-dockerfile.mdx

```diff
@@ -0,0 +1,89 @@
+---
+title: Best practice Dockerfile for Python with poetry
+ogTitle: Best practice Dockerfile for Python with poetry
+description: A sample best practice poetry Dockerfile for Python from Depot
+---
+
+Below is an example `Dockerfile` that we use and recommend at Depot when we are building Docker images for Python applications that use `poetry`as their package manager.
+
+```dockerfile
+FROM python:3.12-slim AS base
+
+FROM base AS builder
+ENV PYTHONUNBUFFERED=1 \
+  POETRY_VERSION=1.6.1 \
+  POETRY_VIRTUALENVS_CREATE=false \
+RUN --mount=type=cache,target=/root/.cache/pip \
+  pip install "poetry==$POETRY_VERSION"
+WORKDIR /app
+COPY pyproject.toml poetry.lock ./
+RUN poetry install --no-dev
```

Comment:
Does this need the cache mount? (I'm not super familiar with what `poetry install` does)

Reply:
I took another look. What makes this difficult is the vastly different number of ways someone could be using poetry.
I just found this (I think) better example than what I was referencing earlier.
https://github.com/michaeloliverx/python-poetry-docker-example/blob/master/docker/Dockerfile

Poetry docs explaining some of the challenges of caching: https://python-poetry.org/docs/faq/#poetry-busts-my-docker-cache-because-it-requires-me-to-copy-my-source-files-in-before-installing-3rd-party-dependencies

It looks like ironically venvs are going to help here by standardizing the cache between package managers.
Just updated the page fairly significantly. This works differently than the last example. There is now no longer a conversion to requirements.txt